### ISSUE TITLE
[help] `make mnist_conv2d_minimal` fail on ubuntu

### ISSUE THREAD

**Issue by tiendung** [fengwang/ceras#3]
When I remove `constexpr` it can compiled. Mine env is Ubuntu, g++ 11.3.0

```
make mnist_conv2d_minimal
g++ -c -std=c++20 -Wall -Wextra -fmax-errors=1 -ftemplate-backtrace-limit=0 -fdata-sections -ffunction-sections -fconcepts-diagnostics-depth=4 -ftemplate-depth=100860 -funsafe-math-optimizations -Ofast -flto=auto  -funroll-all-loops -pipe -march=native   -o ./obj/test_mnist_conv2d_minimal.o test/mnist_conv2d_minimal.cc
In file included from test/../include/./activation.hpp:4,
                 from test/../include/ceras.hpp:6,
                 from test/mnist_conv2d_minimal.cc:1:
test/../include/././operation.hpp: In function ‘constexpr auto ceras::general_conv2d(long unsigned int, long unsigned int, long unsigned int, long unsigned int, const string&)’:
test/../include/././operation.hpp:1698:27: error: invalid return type ‘ceras::general_conv2d(long unsigned int, long unsigned int, long unsigned int, long unsigned int, const string&)::<lambda(const Ex&, const Ey&)>’ of ‘constexpr’ function ‘constexpr auto ceras::general_conv2d(long unsigned int, long unsigned int, long unsigned int, long unsigned int, const string&)’
 1698 |     auto constexpr inline general_conv2d
      |                           ^~~~~~~~~~~~~~
```

**Comment by fengwang** [comment]
In my point of view, this error is related to `std::string` constructor with g++11.3, in which `constexpr basic_string( const basic_string& other );` is not yet well implemented. This error can be reproduced [here](https://godbolt.org/z/x64jP39zE) with g++11.3, but can be solved by switching to g++12.2.

So I would suggest upgrade to g++12.2 if possible. Or remove the `constexpr` specifier for every function that takes `std::string` type argument.

**Comment by tiendung** [comment]
Everthing works fine with g++12, thank you!
